src/config.cpp

```diff
@@ -664,8 +665,23 @@ void ParseConfig(const fs::path& filename, Config& config) {
   }
 }
 
-Config::Config(const fs::path& path) : config_path{path} {
+void ParseConfig(std::string_view json, Config& config) {
```

Comment:
I think you should add the JSON string as a second parameter to ParseConfig, then add this as a second block after it's try/catch. Also change the string slightly:
`Error encountered while parsing config overlay: JSON Error: (message) at line ## index ##"`
Since it will be confusing the way it is currently worded:
`Error encountered while parsing JSON JSON Error: (message) at line ## index ##"`